pyugrid/util.py

```diff
@@ -49,3 +49,28 @@ def _signed_area_tri(points):
     x3, y3 = points[2]
 
     return(((x1-x3)*(y2-y3)-(x2-x3)*(y1-y3))/2)
+
+def asarraylike(obj):
+    """
+    tests if obj acts enough like an array to be used in pyugrid. 
+
+    If it does the object is returned as is. If not, then numpy's
+    array() will be called on it.
+
+    This should catch netCDF4 variables, etc.
+
+    Note: these won't check if the attributes required actually work right.
+
+    :param obj: The object to check if it's like an array
+
+    """
+
+    must_have = ['dtype', 'shape']
```

Comment:
> You mean one of [dask, biggus, ???] -- I don't think so.

OK.  (Warning: I will try to push this again later ;-)

> But the goal was to really do duck-typing -- anything that works is OK to use. 

Got it.

So I say one of:

> - what I've got now, with new methods added as we see the need or
> - just go with array and be done with it.

I prefer the former.  I am OK with adding stuff as we go.